yt_dlp/extractor/mlb.py

```diff
@@ -347,3 +347,69 @@ def _real_extract(self, url):
             'subtitles': subtitles,
             'http_headers': {'Authorization': f'Bearer {self._access_token}'},
         }
+
+
+class MLBArticleIE(InfoExtractor):
+    _VALID_URL = r'https?://www\.mlb\.com/news/(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'https://www.mlb.com/news/manny-machado-robs-guillermo-heredia-reacts',
+        'info_dict': {
+            'id': '36db7394-343c-4ea3-b8ca-ead2e61bca9a',
+            'title': 'Machado\'s grab draws hilarious irate reaction',
+            'modified_timestamp': 1650130737,
+            'description': 'md5:a19d4eb0487b2cb304e9a176f6b67676',
+            'modified_date': '20220416',
+        },
+        'playlist_count': 2,
+    }]
+
+    def get_single_video(self, video_slug, display_id):
+        json_api_data = self._download_json(
+            f'https://www.mlb.com/data-service/en/videos/{video_slug}', display_id)
+
+        formats, subtitles = [], {}
+        for playback_url in traverse_obj(json_api_data, ('playbacks', ..., 'url')) or []:
+            if determine_ext(playback_url) == 'm3u8':
+                fmt, sub = self._extract_m3u8_formats_and_subtitles(playback_url, display_id, ext='mp4')
+                formats.extend(fmt)
+                self._merge_subtitles(sub, target=subtitles)
+            elif determine_ext(playback_url) == 'jpg':
+                continue
+            else:
+                formats.append({
+                    'url': playback_url,
+                    'ext': 'mp4',
+                })
+        self._sort_formats(formats)
+        return {
+            'id': video_slug,
+            'title': json_api_data.get('title'),
+            'description': json_api_data.get('description'),
+            'formats': formats,
+            'subtitles': subtitles,
+            'duration': parse_duration(json_api_data.get('duration')),
+            'timestamp': parse_iso8601(json_api_data.get('timestamp'))
+        }
```

Comment:
Why is this not just being passed through to `MLBVideoIE` with a url_result?

Reply:
Implemented in f9b1885846a0942ff60dd29d915dfc135498999f